### ISSUE TITLE
Issue saving the model as tflite using `tf.lite.TFLiteConverter.from_keras_model` on `tensorflow==2.16.1`

### ISSUE THREAD

**Issue by jirenmaa** [Signa-Lingua/singa-ml#4]
![Screenshot 2024-06-09 220318](https://github.com/Signa-Lingua/singa-ml/assets/68807856/59b3aba7-d2cd-4e75-a37b-db5b34b5b54d)

Encountered an issue while running the code in the Jupyter kernel. The error message is as follows:

```
error: missing attribute 'value'
LLVM ERROR: Failed to infer result type(s).
```

```python
...

def save_as_tflite(_model, model_path):
    converter = tf.lite.TFLiteConverter.from_keras_model(_model)

    converter.target_spec.supported_ops = [
        tf.lite.OpsSet.TFLITE_BUILTINS,  # enable TensorFlow Lite ops.
        tf.lite.OpsSet.SELECT_TF_OPS,  # enable TensorFlow ops.
    ]

    tflite_model = converter.convert()

    with open(model_path, "wb") as f:
        f.write(tflite_model)

    f.close()

def save_model(_model, extension="keras", base_name="singa_slr_v_"):
    model_dir = "../storage/models/" + extension

    next_filename = get_next_filename(model_dir, base_name, extension)
    model_path = os.path.join(model_dir, next_filename)

    match extension:
        case "tflite":
            save_as_tflite(_model, model_path)
            print(f"saved as tflite at {model_path}")

        case "keras":
            save_as_keras(_model, model_path)
            print(f"saved as keras at {model_path}")
```

calling the save function
```python
save_as_tflite(_model, model_path)
```

This issue seems similar to the following issues on the official TensorFlow GitHub repository:

[Issue #69237](https://github.com/tensorflow/tensorflow/issues/69237)
[Issue #62855](https://github.com/tensorflow/tensorflow/issues/62855#issuecomment-2104167729)

**Comment by jirenmaa** [comment]
Found a workaround suggested in [Issue #64515](https://github.com/tensorflow/tensorflow/issues/64515#issuecomment-2025267530). The suggestion is to use [ExportArchive](https://keras.io/api/models/model_saving_apis/export/) to convert a Keras model to a saved model first.

Haven't tested this yet but plan to do so later.

**Comment by jirenmaa** [comment]
It seems like a simple way to fix the issue [Comment #64515](https://github.com/tensorflow/tensorflow/issues/64515#issuecomment-2036195525) is to downgrade the Keras version from 3.x.x to 2.x.x and set the environment to use the legacy Keras, as suggested:

```
pip install -U tf_keras # Keras 2
```

Additionally, some changes to the code might be necessary:

```python
import os

os.environ["TF_USE_LEGACY_KERAS"] = "1"
```

**Comment by jirenmaa** [comment]
Reopening this issue because the previous methods were inconsistent. Currently, I'm experiencing the same issue as described by the user in this [comment #19108](https://github.com/keras-team/keras/issues/19108#issuecomment-1913421572). The suggestion made in the comment is to use `keras.Model.export`, which appears to be working.

For a comprehensive summary of the issue related to converting the model into TFLite on `tensorflow==2.16.1`, please refer to this [comment #62989](https://github.com/tensorflow/tensorflow/issues/62989#issuecomment-1967609913).

Here's an example of the code for converting the model into TfLite with TensorFlow version 2.16.1 from those issues:

```python
import tensorflow as tf

import keras

(train_images, train_labels), (
    test_images,
    test_labels,
) = keras.datasets.fashion_mnist.load_data()

test_model = keras.Sequential(
    [
        keras.layers.Flatten(),
        keras.layers.Dense(64, activation="relu"),
        keras.layers.Dense(10, activation="softmax"),
    ]
)
test_model.compile(
    optimizer="adam",
    loss=keras.losses.SparseCategoricalCrossentropy(),
    metrics=["accuracy"],
)

test_model.fit(train_images, train_labels, epochs=1)

test_model.export("model_test")

converter = tf.lite.TFLiteConverter.from_saved_model("model_test")
tflite_model = converter.convert()

with open("model.tflite", "wb") as f:
    f.write(tflite_model)

f.close()
```
